llama-index-integrations/embeddings/llama-index-embeddings-cohere/llama_index/embeddings/cohere/base.py

```diff
@@ -74,14 +74,14 @@ class CohereEmbedding(BaseEmbedding):
     # Instance variables initialized via Pydantic's mechanism
     cohere_client: Any = Field(description="CohereAI client")
     truncate: str = Field(description="Truncation type - START/ END/ NONE")
-    input_type: Optional[str] = Field(description="Model Input type")
+    # input_type: Optional[str] = Field(description="Model Input type")
 
     def __init__(
         self,
         cohere_api_key: Optional[str] = None,
-        model_name: str = "embed-english-v2.0",
+        model_name: str = "embed-english-v3.0",
         truncate: str = "END",
-        input_type: Optional[str] = None,
```

Comment:
if we remove this, it might introduce break changes bc other places might still pass it

Reply:
ya, should still allow an override

Reply:
would those other places be using it by just instantiating the `embed_model` and then passing it around to be used, like below? 
```python
# Create the embedding model
embed_model = CohereEmbedding(
    cohere_api_key="{API_KEY}",
    model_name="embed-english-v3.0",
    input_type="search_query",
)

# Create the service context with the cohere model for generation and embedding model
service_context = ServiceContext.from_defaults(
    llm=Cohere(api_key="{API_KEY}", model="command"),
    embed_model=embed_model
)
```
If so, I can look for the places that reference this and update the examples that init the `embed_model` in this way. 

If instead there are e2e workflows that init the CohereEmbed model and pass the search param that we know of that would be problematic to change, maybe we can keep the `input_type` param and raise a warning that it will be removed in upcoming releases or something like that? wdyt @hatianzhang ?